Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+    else
+        json.range = '';
+    end
+
+    if isnumeric(TempValue)
+        TempValue = num2str(TempValue);
+    end
+
+    if size(TempValue, 1) == 1
+        TextString = sprintf([sprintf('%-20s',[json.alias,':']), sprintf('%8s',TempValue),sprintf('%-12s', [json.unit, ' ' , json.range]), ' \n']);
+    end
+
+    line = xASL_sub_PrintText(TextString, figure, line, settings);
+
+end
+
+function [settings] = xASL_sub_PrintImage(input, x, figure, settings, position)
+    switch nargin
+    case 4
+        position = [xASL_str2num(input.position) xASL_str2num(input.size)];
+
+        if isfield(input, 'xPath') && isfield(x.P, input.name)
+            ImagePath = x.P.(input.name);
+        elseif isfield(input, 'absolutePath')
+            ImagePath = input.absolutePath;
+        elseif isfield(input, 'popPath')
+            ImagePath = fullfile(x.dir.xASLDerivatives, 'Population', input.popPath);
+            ImagePath = xASL_sub_replace(ImagePath, x);
+        elseif isfield(input, 'subjPath')
+            ImagePath = fullfile(x.dir.xASLDerivatives, x.SUBJECT, input.subjPath);
+        else
+            warning('xASL_sub_PrintImage didnt have a defined path') ;
+            return
+        end
+
+        if isfield(input, 'header')
+            header = input.header;
+        else
+            header = '';
+        end
+    case 5
+        ImagePath = input;
+        header = '';
+    end
+
+    [canvas, ~] = xASL_sub_createNewCanvas(position(1:2), position(3:4), settings.canvas);
+    ax=axes('Position', canvas, 'Visible', settings.axesVisible, 'Parent', figure);
+    [img, ~, alphachannel] = imread(ImagePath);
+    fg= imshow(img);
+
+    if strcmp('png', ImagePath(end-2:end))
+        fg.AlphaData=alphachannel;
+    end
+
+    clear fg
+    settings.figurecount = xASL_sub_PrintHeader(header, figure, settings, canvas);
+end
+
+function [strout] = xASL_sub_replace(strin, x)
+    strout = strin;
+    substring = regexp(strin, '<\w*>', 'match');
+    for substringIndex=1:length(substring)
+        if ~isfield(x, substring{substringIndex}(2:end-1))
+            warning(['Could not replace ', substring{substringIndex}, ' check if file exists in ExploreASL/Derivatives/Population']);
+        else
+            strout = strrep(strout, substring{substringIndex}, x.(substring{substringIndex}(2:end-1)));
+        end
+    end
+end
+
+function [settings] = xASL_sub_PrintScan(input, x, figure, settings)
+    position = [xASL_str2num(input.position) xASL_str2num(input.size)];
+    [canvas, ~] = xASL_sub_createNewCanvas(position(1:2), position(3:4), settings.canvas);
+
+    if ~isfield(x.P, input.name)
+        warning (['could not print', input.name, 'check if nifti exists in ExploreASL/Derivatives/Population']);
+        return
+    end
+
+    ImIn = {x.P.(input.name)};
+    header = input.name;
+    ax=axes('Position', canvas, 'Visible', settings.axesVisible, 'Parent', figure);
+    
+    if isfield(input, 'overlay')
```

Comment:
I don't understand, a slice is a subset of a full image, and thus of an overlay. Does that mean that this function has multiple functions depending on its input? Because that would be confusing... Unless you have a single function that handles all inputs, but now it only handles overlay and slice inputs, we should avoid functions becoming convoluted.